### ISSUE TITLE
Warn about overloads that make other overloads useless

### ISSUE THREAD

**Issue by twoertwein** [PyCQA/flake8-pyi#246]
The same example as in #245 but with one overload that I omitted in #245 (the last overload):

```py
    @overload
    def agg(self, arg: str, *args, **kwargs) -> DataFrame: ...
    @overload
    def agg(self, arg: Dict, *args, **kwargs) -> DataFrame: ...
    @overload
    def agg(self, arg: FuncType, *args, **kwargs) -> DataFrame: ...
    @overload
    def agg(self, *args, **kwargs) -> DataFrame: ...
```

While the last overload allows specifying no positional arguments, it also makes the other overloads obsolete: the first positional parameter (if present) can be of any type, not only of `str | Dict | FuncType`.

**Comment by JelleZijlstra** [comment]
This feels like it might be a better fit for a type checker, not a linter, since it requires a fairly sophisticated understanding of overloads.

**Comment by AlexWaygood** [comment]
I agree with @JelleZijlstra: this one feels out of scope. You're getting into the question of "Is parameter specification X a subtype of parameter specification Y?", and that feels like something a type checker is far better equipped to answer than a humble linter :)
